Replicator/DBAccess.cc

```diff
@@ -63,6 +64,8 @@ namespace litecore { namespace repl {
                     if (!idb) {
                         logError("Couldn't open new db connection: %s", c4error_descriptionStr(error));
                         idb = c4db_retain(db);
+                    } else {
+                        c4db_setDatabaseTag(idb, DatabaseTagReplicator);
```

Comment:
As far as I can tell this is the only db that gets tagged Replicator. But this db is only used to insert pulled docs. So the main replicator database isn't tagged. It isn't created by DBAccess; IIRC it's created by C4Replicator and passed into the main Replicator.

Reply:
Look for the call to `db->openAgain()` -- that's what opens it.

Come to think of it, you should make openAgain propagate the tag to the new database.

Reply:
I grepped all c4db_open in the source and ignored places in files that has Test. 
If the app opens a db, with c4db_openX(), and we, for our own purpose, like the one here, openAgain(). Customers still owns the db they created, right?

Reply:
Sorry, only grepped c/cc files. There are some header doing it.
Will fix them

Reply:
Found two new places, c4IncomingReplicator.hh, c4RemoteReplicator.hh
There is still failed case (a new case) with FAIL-FAST.